flake.nix

```diff
@@ -16,13 +16,80 @@
       system:
       let
         pkgs = nixpkgs.legacyPackages.${system};
+
```

Comment:
The  purpose of this file to allow devs who use Nix or NixOS to spin up a reproducable dev environment with all the required apps/libs needed to work with this repo. Devs can choose to use it or not. Doesn't have any effect on the repo itself. Even if you use MacOS its possible to benefit from Nix but its totally optional. The concept is explained here if you're interested https://www.youtube.com/watch?v=0YBWhSNTgV8